docs/aks/demo/DEMO_SCRIPT.md

```diff
@@ -0,0 +1,389 @@
+helm3 repo add aad-pod-identity https://raw.githubusercontent.com/Azure/aad-pod-identity/master/charts
+helm3 repo add application-gateway-kubernetes-ingress https://appgwingress.blob.core.windows.net/ingress-azure-helm-package/
+helm3 repo update
+```
+
+## Install Linkerd Service Mesh into the cluster
+
+Download the Linkerd v2 CLI:
+
+```shell
+# macOS and linux/WSL
+
+curl -sL https://run.linkerd.io/install | sh
+export PATH=$PATH:$HOME/.linkerd2/bin
+```
+
+Install the Linkerd control plane in the linkerd namespace:
+
+```shell
+linkerd install | kubectl apply -f -
+```
+
+Validate the install with:
+
+```shell
+linkerd check
+```
+
+## Install the NGNIX ingress controller
+
+Create a namespace for your ingress resources. There is a yaml file located in the clones repository under ./docs/aks/cluster/manifests/ingress-nginx-namespace.yaml
+
+```shell
+kubectl apply -f cluster/manifests/ingress-nginx-namespace.yaml
+```
+
+Use Helm to deploy an NGINX ingress controller
+
+```shell
+helm3 install ingress stable/nginx-ingress \
+    --namespace ingress-nginx \
+    --set controller.replicaCount=2 \
+    --set controller.nodeSelector."beta\.kubernetes\.io/os"=linux \
+    --set defaultBackend.nodeSelector."beta\.kubernetes\.io/os"=linux
+```
+
+## Deploy the needed componenets of helium, key rotator and the testing harness
+
+An helm chart is included for the reference application ([helium](https://github.com/RetailDevCrews/helium-csharp))
+
+Install the Helm Chart located in the cloned directory
+
+```shell
+cd docs/aks/cluster/charts
+```
+
+Create a file called helm-config.yaml with the following contents that should be edited to fit the environment being deployed in
+
+```yaml
+# Default values for bluebell.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+labels:
+  aadpodidbinding: podid #should be value of $LABEL from the output o aad-podid.sh
+
+image:
+  repository: xxxx.azurecr.io #The spceific acr created for this environment
+  name: helium-csharp # The name of the image for the helium-csharp repo
+
+annotations:
+  linkerd.io/inject: enabled # Allows for the application to be injected into the Service Mesh
+
+ingress:
+  hosts:
+    - host: 52.230.216.170.nip.io # Replace the IP address with the IP of the nginx external IP. kubectl get svc -n ingress-nginx to see the correct IP
+      paths: /
+
+keyVaultName: helium-aks-kv # Replace with the name of the Key Vault that holds the secrets
+```
+
+This file can now be given to the the helm install as an override to the default values.
+
+```shell
+helm3 install helium-aks . --set image.repository=<acr_name>.azurecr.io -f helm-config.yaml
+```
+
+## Dashboard setup
+
+Open `AKS Dashboard.json` in an editor and update `{SUBSCRIPTION_GUID}` with your subscription, `{AKS_RESOURCE_GROUP}` with your AKS cluster resource group name, and `{COSMOS_RESOURCE_GROUP}` with your Cosmos resource group name.
```

Comment:
can we make the changes with sed and env vars?

Reply:
Lets make this non blocking for PR and will move to M5 for cleanup